src/resultsdisassemblypage.cpp

```diff
@@ -116,25 +134,61 @@ void ResultsDisassemblyPage::showDisassembly(const QString& processName, const Q
         int row = 0;
         m_model->clear();
 
-        m_model->setHorizontalHeaderLabels({tr("Assembly")});
+        QStringList headerList;
+        headerList.append({tr("Assembly")});
+        for (int i = 0; i < m_callerCalleeResults.selfCosts.numTypes(); i++) {
+            headerList.append(m_callerCalleeResults.selfCosts.typeName(i));
+        }
+        m_model->setHorizontalHeaderLabels(headerList);
 
+        int numTypes = m_callerCalleeResults.selfCosts.numTypes();
         QTextStream stream(&m_tmpFile);
         while (!stream.atEnd()) {
             QString asmLine = stream.readLine();
             if (asmLine.isEmpty() || asmLine.startsWith(QLatin1String("Disassembly"))) continue;
 
+            const auto asmTokens = asmLine.splitRef(QLatin1Char(':'));
+            const auto addrLine = asmTokens[0].trimmed();
+
+            bool ok = false;
+            const auto addr = QStringRef(addrLine).toULongLong(&ok, 16);
```

Comment:
the explicit QStringRef here isn't needed, addrLine is a QStringRef already

Reply:
I don't see initial question about addTab but as I remember it was your concern about output uniqueness. I confirm that the output is proper and unique for each onJumpToDisassembly and addTab. And before any symbol selected I prefer not to show the tab.